diff --git a/src/Console/Command/Compile.php b/src/Console/Command/Compile.php
index ebc1ee6..73e8b62 100644
--- a/src/Console/Command/Compile.php
+++ b/src/Console/Command/Compile.php
@@ -546,7 +546,7 @@ HELP;
             );
 
             if (is_callable($metadata)) {
-                $metadata = $metadata();
+                $metadata = $metadata($config, $box);
             }
 
             $logger->log(
